### ISSUE TITLE
Research and Implement simplified way to add Brython to a page

### ISSUE THREAD

**Issue by juancarlospaco** [brython-dev/brython#69]
- Research and Implement a simplified way to add Brython to a page.
- 1 line added to add Brython to a existing page. K.I.S.S.
- Move the onLoad  somewhere?, bind a callback?, on url parameter?, something else?
- Update Docs to reflect that change. Add Async and Defer script tag attributes to Docs?

Context:
https://github.com/brython-dev/brython/commit/5c5c44e905da4560712118f630ad5566a65fbb72#commitcomment-8631942

https://github.com/brython-dev/brython/commit/e770f9e55b1a8647e532295a300b59dd712852a5

:pensive:

**Comment by earney** [comment]
can we close this issue?  Has it been addressed?

**Comment by juancarlospaco** [comment]
So the Final Version should be a one-liner:

```javascript
(function(){var _s=document.createElement('script');_s.src="http://brython.info/src/brython_dist.js";_s.onload=function(){brython({ debug:1}) };document.head.appendChild(_s);})();
```

Or expanded:

```javascript
(function() {
    var _s = document.createElement('script');
    _s.src = "http://brython.info/src/brython_dist.js";
    _s.onload = function() { brython({ debug: 1 }) };
    document.head.appendChild(_s);
})();
```

:question:

**Comment by earney** [comment]
or place the "one-liner" in a file and import it with a script tag, like so:

<script type="type/javascript"  src="one-liner.js"></script>

**Comment by PierreQuentel** [comment]
How do you pass arguments to the function brython() in this case ?

**Comment by earney** [comment]
How about this:

<script type="text/javascript"
data-brython-options="debug:1;variable2:blah"></script>

**Comment by juancarlospaco** [comment]
And as URL parameters ?

**Comment by earney** [comment]
I have added an implementation of this to the repo:

See file: /site/one-liner.html , for an example.

This html file uses a new file called /src/one-liner.js, which contains some of the code (described) above, but also adjusts to higher debug values to put script tags of all the py_*.js files instead of the one file brython_dist.js.  The script tag that "imports" one-liner.js can contain an attribute "data-brython-options" which accepts a string (a json parseable string) ,and converts it into an object that the brython function accepts after page load.

Billy

**Comment by earney** [comment]
Has anyone taken a look at this?   Any comments?

**Comment by juancarlospaco** [comment]
I  look at this. I leave comments.  :smile:

**Comment by juancarlospaco** [comment]
What do you think of also providing a tiny .html file to use with HTML Import :grey_question: 
http://caniuse.com/#feat=imports Webkit, Chrome, Android, Firefox can use it now.

**Comment by kikocorreoso** [comment]
@juancarlospaco from the link you provided it seems FF is not supporting that and they do not have plans to do it... :disappointed:

**Comment by earney** [comment]
I have pushed some changes to brython_loader.js. 

I have decided to create two attributes in the script tag that loads brython_loader.js:
data-loader-options and data-brython-options,  Both of these take a json string (and will be parsed by JSON.parse).  What ever is passed to data-brython-options will be parsed by JSON.parse and the resultant object is passed to the brython function, which is called on page load.

data-loader-options is also a json string (will be parsed by JSON.parse), and will be used to figure out how load brython source code.  the debug variable describes if we should load brython.js, or the individual brython files (py_*.js).  If we set VFS to true, we also load py_VFS.js.   There is also a dist variable, if set to true, will load the brython_dist.js file (and this will ignore all other passed options). 

Here's an example which is taken from sites/one-liner.html:

<script type="text/javascript" src="/src/brython_loader.js"
data-loader-options='{"VFS": true, "debug":10}'
data-brython-options='{"debug":1,"cache":"none"}'></script>
